manul/src/combinators/misbehave.rs

```diff
@@ -0,0 +1,269 @@
+use alloc::{
+    boxed::Box,
+    collections::{BTreeMap, BTreeSet},
+};
+use core::fmt::Debug;
+
+use rand_core::CryptoRngCore;
+
+use crate::protocol::{
+    Artifact, BoxedRng, BoxedRound, Deserializer, DirectMessage, EchoBroadcast, EntryPoint, FinalizeError,
+    FinalizeOutcome, LocalError, NormalBroadcast, ObjectSafeRound, PartyId, Payload, ReceiveError, RoundId, Serializer,
+};
+
+/// A trait describing required properties for a behavior type.
+pub trait Behavior: 'static + Debug + Send + Sync {}
+
+impl<T: 'static + Debug + Send + Sync> Behavior for T {}
+
+/// The new entry point for the misbehaving rounds.
+///
+/// Use as an entry point to run the session, with your ID, behavior `B` and the misbehavior definition `M` set.
+#[derive_where::derive_where(Debug)]
+pub struct MisbehavingEntryPoint<Id, B, M>
+where
+    Id: PartyId,
+    B: Behavior,
+    M: Misbehaving<Id, B>,
+{
+    round: BoxedRound<Id, <M::EntryPoint as EntryPoint<Id>>::Protocol>,
+    behavior: Option<B>,
+}
+
+/// A trait defining a sequence of misbehaving rounds amending the messages sent by some existing ones.
```

Comment:
```suggestion
/// A trait defining a sequence of misbehaving rounds changing the messages sent by some existing, well behaved, rounds.
```

The word "amend" usually means to "correct" something or "make more accurate", so that clashes a bit with the misbehaviour context here. "Change" is pretty neutral; "manipulate" is malicious. Maybe there are more terms we could consider.

Reply:
Changed to "modify", but that's neutral. Perhaps we should pick something with an "evil" connotation.